src/main/java/com/taboola/rest/api/internal/CommunicationFactory.java

```diff
@@ -35,7 +35,7 @@ private HttpLoggingInterceptor createLoggingInterceptor(CommunicationConfig conf
             loggingInterceptor.redactHeader("Authorization");
             loggingInterceptor.redactHeader("Cookie");
         } else {
-            loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BASIC);
+            loggingInterceptor.setLevel(config.getLoggingLevel());
```

Comment:
This is creating an issue. You can set BODY and in such case you not redacting sensitive headers like first if block